test_bench.py

```diff
@@ -73,12 +73,16 @@ class TestBenchNetwork:
     """
     def test_train(self, hub_model, benchmark):
         try:
+            (model, inputs) = hub_model.get_module()
```

Comment:
my only concern with this approach is if we know that _every_ model uses the thing that is returned from .get_module() as the pytorch model object that's used for training. 

iirc there was at least one case where the train function and eval function actually construct different model objects, and bc of this i think the thing that get_module returns could be not used for training in some instances.  have you checked this?

Reply:
as for alternatives, 

1) putting model .train() or .eval() call inside of the benchmark_model.train() and .eval() functions would annoyingly include it in the time measurement, which you probably wanted to avoid...

2) we could construct a .train_model and .eval_model in __init__ for each benchmark,  and ensure they are pre-configured correctly.  but this would use more memory,  might break some models on gpu..

3) we could add an API to the benchmark_model wrapper, set_mode(train/eval) which lets each model choose whether to simply call .eval()/.train() on a shared model, whether to maintain 2 copies of the model, etc.

Reply:
> get_module returns could be not used for training in some instances. have you checked this?

I checked all vision models and attention, but I will check the rest.

> 1. 

Exactly, that's why I wanted to keep it outside `benchmark` calls.

> 2.

Yes, there could memory issues.

> 3. 

I like this one, I think, if I understand it correctly :-) I will have to go through all the models anyways, might as well add `set_mode` to all of them. I didn't quite get the `benchmark_model` wrapper point (since I couldn't find the base extension point) I'll ping soon to clarify